### PR TITLE
Add annotation endpoints

### PR DESCRIPTION
**Pull request by alpkaanusen**

### CODE REVIEW

File: backend/annotation/views.py [hunk 1]
```diff
@@ -1,3 +1,67 @@
...
+                body = request_data['body'] if 'body' in request_data else None
+                annotation = Annotation()
+                annotation.context = request_data['context']
+                annotation.IRI = request_data['id']
+                annotation.motivation = request_data['motivation']
+                annotation.creator = request_data['creator']
+                annotation.save()
+
+                if body:
+                    body_annotation = Annotation.objects.filter(IRI=request_data['id'])
```
[the remark this answers is not in the document]
Comment (alpkaanusen) [reply backend/annotation/views.py]: We can change the ReferenceField in Body to a ListField.

[the remark this answers is not in the document]
Comment (emingure) [reply backend/annotation/views.py]: Instead we can use the annotation object created above so problem solves, ?

File: backend/annotation/views.py [hunk 2]
```diff
@@ -17,47 +17,45 @@ def annotation_handler(request):
...
             try:
                 user_id = authentication.get_user_id(token)
-                request_data = json.loads(request.body.decode('utf-8'))
+                request_data = request.data
                 targets = []
                 if 'targets' in request_data:
                     targets = request_data['targets']
                 body = request_data['body'] if 'body' in request_data else None
                 annotation = Annotation()
                 annotation.context = request_data['context']
```
Comment (emingure) [inline backend/annotation/views.py]: we need to check if request data has context

File: backend/annotation/views.py [hunk 3]
```diff
@@ -17,47 +17,45 @@ def annotation_handler(request):
...
+            annotations = Annotation.objects.filter(annotation_object__IRI=query)
             return JsonResponse({"response": True, "annotations": annotations})
         except Exception as e:
             return JsonResponse({'response': False, 'error': str(e)})
     elif request.method == 'POST':
         if token and authentication.is_authenticated(token):
             try:
                 user_id = authentication.get_user_id(token)
-                request_data = json.loads(request.body.decode('utf-8'))
+                request_data = request.data
```
Comment (emingure) [inline backend/annotation/views.py]: Are you sure it works?

File: backend/annotation/views.py [hunk 4]
```diff
@@ -25,16 +25,16 @@ def annotation_handler(request):
...
                 user_id = authentication.get_user_id(token)
-                request_data = request.data
+                request_data = json.loads(request.body.decode('utf-8'))
                 targets = []
                 if 'targets' in request_data:
                     targets = request_data['targets']
                 body = request_data['body'] if 'body' in request_data else None
                 annotation = Annotation()
-                annotation.context = request_data['context']
+                annotation.context = request_data['context'] if 'context' in request_data else None
```
Comment (emingure) [inline backend/annotation/views.py]: Let's not say ``None`` to it so it takes default value if not specified.

File: backend/annotation/views.py [hunk 5]
```diff
@@ -25,16 +25,16 @@ def annotation_handler(request):
...
                 annotation.IRI = request_data['url']
                 annotation.motivation = request_data['motivation']
-                annotation.creator = request_data['creator']
+                annotation.creator = request_data['creator'] if 'creator' in request_data else None
```
Comment (emingure) [inline backend/annotation/views.py]: Please take creator from auth token

File: backend/api/utils.py [hunk 6]
```diff
@@ -243,4 +243,40 @@ def handle_uploaded_file(app_name, file, filename):
 
     with open(cumulative + '/' + filename, 'wb+') as destination:
         for chunk in file.chunks():
-            destination.write(chunk)
\ No newline at end of file
+            destination.write(chunk)
+
+
+def annotation_json(annotation):
```
Comment (emingure) [inline backend/api/utils.py]: We need to call other json funcs below from here so we have a json response fit to w3c standard

Comment (emingure) [reply backend/api/utils.py]: Or ignore the comment above and remove this function as we have ```def annotation_object(self):``` func in model

Comment (alpkaanusen) [reply backend/api/utils.py]: Note that annotation_object(self) would not work the same way without annotation_json(annotation), and there would be more redundancy

File: backend/annotation/views.py (hunk 4 above)
Comment (alpkaanusen) [reply backend/annotation/views.py]: context currently does not have a default value

File: backend/annotation/views.py (hunk 5 above)
Comment (alpkaanusen) [reply backend/annotation/views.py]: Changed

File: backend/annotation/views.py (hunk 4 above)
Comment (emingure) [reply backend/annotation/views.py]: Please add this: ``http://www.w3.org/ns/anno.jsonld``

File: backend/annotation/models.py [hunk 7]
```diff
@@ -42,8 +42,11 @@ def annotation_object(self):
         obj['IRI'] = self.IRI
         obj['motivation'] = self.motivation
         obj['creator'] = self.creator
-        target = Target.objects.get(annotation=self)
-        obj['target'] = target_json(target)
+        targets = Target.objects.filter(IRI=self.IRI)
```
Comment (emingure) [inline backend/annotation/models.py]: Please use the previous one. There may be multiple targets for the same url but they may be related to different annotations

**Pull request merged**